indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -0,0 +1,171 @@
+import {
+  TooltipIfTruncated,
+  ManagementPageSubTitle,
+  ManagementPageBackButton,
+} from 'indico/react/components';
+import {useNumericParam} from 'indico/react/util/routing';
+import {Translate} from 'indico/react/i18n';
+import {EditableType} from '../../models';
+import StateIndicator from '../../editing/timeline/StateIndicator';
+import './EditableList.module.scss';
+
+export default function EditableList() {
+  const list = []; // Replace it with list from server
+  const eventId = useNumericParam('confId');
+  const {type} = useParams();
+  const [sortBy, setSortBy] = useState('id');
+  const [sortDirection, setSortDirection] = useState('ASC');
+  const [sortedList, setSortedList] = useState(list); // list from server
+  const [checked, setChecked] = useState([]);
+  const title = {
+    [EditableType.paper]: Translate.string('List of papers'),
+    [EditableType.slides]: Translate.string('List of slides'),
+    [EditableType.poster]: Translate.string('List of posters'),
+  }[type];
+  const columnHeaders = [
+    ['id', Translate.string('ID'), 60],
+    ['code', Translate.string('Code'), 80],
+    ['title', Translate.string('Title'), 600],
+    ['status', Translate.string('Status'), 150],
+    ['assigned', Translate.string('Assigned'), 400],
+  ];
+
+  // eslint-disable-next-line no-shadow
+  const _sortList = ({sortBy, sortDirection}) => {
+    const newList = _.sortBy(list, [sortBy]);
+    if (sortDirection === SortDirection.DESC) {
+      newList.reverse();
+    }
+    return newList;
+  };
+
+  // eslint-disable-next-line no-shadow
+  const _sort = ({sortBy, sortDirection}) => {
+    setSortBy(sortBy);
+    setSortDirection(sortDirection);
+    setSortedList(_sortList({sortBy, sortDirection}));
+  };
+
+  // eslint-disable-next-line react/prop-types
+  const _cellRenderer = ({dataKey, rowIndex}) => {
+    return (
+      <>
+        {dataKey === 'assigned' && <Icon name="user outline" />}
+        <TooltipIfTruncated>
+          <div index={rowIndex} styleName="rowcolumn-tooltip" role="gridcell">
+            {dataKey === 'status' ? (
+              <StateIndicator index={rowIndex} state="needs_submitter_changes" circular label />
+            ) : (
+              sortedList[rowIndex][dataKey]
+            )}
+          </div>
+        </TooltipIfTruncated>
+      </>
+    );
+  };
+
+  const _onChangeHeaderCheckbox = data => {
+    data.checked ? setChecked(sortedList.map(row => row.id)) : setChecked([]);
+  };
+
+  const _onChangeRowCheckbox = data => {
+    const newRow = sortedList[data.index].id;
+    checked.includes(newRow)
+      ? setChecked(old => old.filter(row => row !== newRow))
+      : setChecked(old => [...old, newRow]);
+  };
+
+  return (
+    <>
+      <ManagementPageSubTitle title={title} />
+      <ManagementPageBackButton url={dashboardURL({confId: eventId})} />
+      <div styleName="editable-topbar">
+        <div>
+          <Button disabled={checked.length === 0} content={Translate.string('Assign')} />
```

Comment:
what about a `const hasCheckedRows = ...` and using that in all the places where you check the length atm?